clang/lib/Driver/Driver.cpp

```diff
@@ -7446,8 +7446,27 @@ void Driver::BuildActions(Compilation &C, DerivedArgList &Args,
       if (Args.hasArg(options::OPT_fsycl_link_EQ) &&
           Args.hasArg(options::OPT_fintelfpga)) {
         // Wrap the object when creating an FPGA AOCX or AOCR binary.
-        auto *BC = C.MakeAction<OffloadWrapperJobAction>(LI, types::TY_LLVM_BC);
-        auto *ASM = C.MakeAction<BackendJobAction>(BC, types::TY_PP_Asm);
+        // When the input file is an AOCR (early) archive, the unbundled host
+        // binary consists of a list of objects. We cannot directly wrap that
+        // binary to be consumed later - this has to go through each listed
+        // object.
```

Comment:
`We cannot directly wrap that binary to be consumed later - this has to go through each listed object.`

For my understanding: what does "each listed object" refer to here?

Reply:
The file that is being 'wrapped' here is a list of files represented by a `TY_Tempfilelist` file.    The contents of that file is what we are actually wrapping (which is a list of objects)